libres/lib/res_util/matrix.cpp

```diff
@@ -1,556 +1,56 @@
-                   __func__);
-}
-
-UTIL_SAFE_CAST_FUNCTION(matrix, MATRIX_TYPE_ID)
-
-/*
-   The matrix objecty is NOT ready for use after this function.
-*/
-static matrix_type *matrix_alloc_empty() {
-    matrix_type *matrix = (matrix_type *)util_malloc(sizeof *matrix);
-    UTIL_TYPE_ID_INIT(matrix, MATRIX_TYPE_ID);
-    matrix->name = NULL;
-    return matrix;
-}
-
-/*
-  The freshly allocated matrix is explicitly initialized to zero. If
-  the variable safe_mode equals true the function will return NULL if
-  the allocation of data fails, otherwise it will abort() if the
-  allocation fails.
-*/
-static matrix_type *matrix_alloc_with_stride(int rows, int columns,
-                                             int row_stride, int column_stride,
-                                             bool safe_mode) {
-    matrix_type *matrix = NULL;
-    if ((rows > 0) && (columns > 0)) {
-        matrix = matrix_alloc_empty();
-        matrix->data = NULL;
-        matrix->data_size = 0;
-        matrix_init_header(matrix, rows, columns, row_stride, column_stride);
-        matrix->data_owner = true;
-        matrix_realloc_data__(matrix, safe_mode);
-        if (safe_mode) {
-            if (matrix->data == NULL) {
-                /* Allocation failed - we return NULL */
-                matrix_free(matrix);
-                matrix = NULL;
-            }
-        }
-    }
-    return matrix;
-}
-
-void matrix_set_name(matrix_type *matrix, const char *name) {
-    matrix->name = util_realloc_string_copy(matrix->name, name);
-}
-
-/*
-   This function will allocate a matrix object where the data is
-   shared with the 'src' matrix. A matrix allocated in this way can be
-   used with all the matrix_xxx functions, but you should be careful
-   when exporting the data pointer to e.g. lapack routines.
-*/
-
-matrix_type *matrix_alloc_shared(const matrix_type *src, int row, int column,
-                                 int rows, int columns) {
-    if (((row + rows) > src->rows) || ((column + columns) > src->columns))
-        util_abort("%s: Invalid matrix subsection src:[%d,%d]  Offset:[%d,%d]  "
-                   "SubSize:[%d,%d] \n",
-                   __func__, src->rows, src->columns, row, column, rows,
-                   columns);
-
-    {
-        matrix_type *matrix = matrix_alloc_empty();
-
-        matrix_init_header(matrix, rows, columns, src->row_stride,
-                           src->column_stride);
-        matrix->data = &src->data[GET_INDEX(src, row, column)];
-        matrix->data_owner = false;
-
-        return matrix;
-    }
-}
-
-static matrix_type *matrix_alloc__(int rows, int columns, bool safe_mode) {
-    return matrix_alloc_with_stride(
-        rows, columns, 1, rows,
-        safe_mode); /* Must be the stride (1,rows) to use the lapack routines. */
-}
 
 matrix_type *matrix_alloc(int rows, int columns) {
-    return matrix_alloc__(rows, columns, false);
+    if (rows <= 0 || columns <= 0)
+        return nullptr;
```

Comment:
Cant see why this should not be an exception instead?

Reply:
Because we have a test that checks this specific case and I don't to change tested behaviour in a refactor PR.